File: src/main/java/com/github/Doomsdayrs/Jikan4java/types/Main/Person/PersonPage/PersonPagePerson.java
package com.github.Doomsdayrs.Jikan4java.types.Main.Person.PersonPage;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.github.Doomsdayrs.Jikan4java.types.Main.Person.Person;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;

import java.io.IOException;
import java.util.ArrayList;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionException;

/**
 * This file is part of Jikan4java.
 * Jikan4java is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * Foobar is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * You should have received a copy of the GNU General Public License
 * along with Jikan4java.  If not, see <https://www.gnu.org/licenses/>.
 * ====================================================================
 * Jikan4java
 * 04 / November / 2018
 *
 * @author github.com/doomsdayrs
 */
public class PersonPagePerson {
    @JsonProperty("mal_id")
    public int mal_id;
    @JsonProperty("url")
    public String url;
    @JsonProperty("image_url")
    public String iconURL;
    @JsonProperty("name")
    public String name;
    @JsonProperty("alternative_names")
    public ArrayList<String> alternative_names;
    
    /**
     * Returns the Person object of this object
     *
     * @return Person Object
     * @throws IOException IOException
     * @throws ParseException ParseException
     */
    public CompletableFuture<Person> getPerson() {
        return CompletableFuture.supplyAsync(() -> {
            try {
                return new ObjectMapper().readValue(((JSONObject) new JSONParser().parse(new OkHttpClient().newCall(new Request.Builder().url("api.jikan.moe/v3/manga/" + mal_id).build()).execute().body().string())).toJSONString(), Person.class);
            } catch (IOException | ParseException e) {
                throw new CompletionException(e);
            }
        });
    }

    @Override
    public String toString() {
        return "PersonPagePerson{" +
                "mal_id=" + mal_id +
                ", url='" + url + '\'' +
                ", iconURL='" + iconURL + '\'' +
                ", name='" + name + '\'' +
                ", alternative_names=" + alternative_names +
                '}';
    }
}

File: src/main/java/com/github/doomsdayrs/jikan4java/data/model/main/person/personPage/PersonPagePerson.kt
package com.github.doomsdayrs.jikan4java.data.model.main.person.personPage

import com.fasterxml.jackson.annotation.JsonProperty
import com.github.doomsdayrs.jikan4java.data.model.main.person.Person
import com.github.doomsdayrs.jikan4java.data.model.support.searchResults.IndividualsPage
import java.util.*

/*
 * This file is part of Jikan4java.
 *
 * Jikan4java is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jikan4java is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jikan4java.  If not, see <https://www.gnu.org/licenses/>.
 * ====================================================================*/
/**
 * Jikan4java
 * 04 / November / 2018
 *
 * @author github.com/doomsdayrs
 */
data class PersonPagePerson(
		@JsonProperty("mal_id") override val mal_id: Int,
		@JsonProperty("url") override val url: String?,
		@JsonProperty("image_url") override val image_url: String?,
		@JsonProperty("name") override val name: String?,
		@JsonProperty("alternative_names") override val alternative_names: ArrayList<String>?
) : IndividualsPage() {
	/**
	 * Returns the Person object of this object
	 *
	 * @return Person Object
	 */
	val person by lazy { retrieve<Person>("$baseURL/person/$mal_id") }
}